### PR TITLE
Fixing manager upgrade failures

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: mgmt_upgrade_test.py [hunk 1]
```diff
@@ -118,23 +110,24 @@ def test_upgrade(self):  # pylint: disable=too-many-locals,too-many-statements
         validate_previous_task_details(task=repair_task, previous_task_details=repair_task_current_details)
         validate_previous_task_details(task=backup_task, previous_task_details=backup_task_current_details)
 
-        with self.subTest("Restoring a 2.0 backup task with 2.1 manager"):
+        with self.subTest("Restoring an older version backup task with newer version manager"):
```
Comment (Orenef11) [inline mgmt_upgrade_test.py:113]: If you can, if you have access to this information, please add the manager version to the `subTest` message.

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1268,12 +1268,13 @@ def is_manager_server_up(self, port=None):
         http_status_code = int(curl_output.stdout.strip())
         return http_status_code == 204
 
-    def wait_manager_server_up(self, verbose=True, timeout=300):
+    def wait_manager_server_up(self, verbose=True, timeout=300, port=None):
```
Comment (Orenef11) [inline sdcm/cluster.py:1271]: I don't see any used with the `port` variable...

**Review by Orenef11 — CHANGES_REQUESTED** [review]

**Comment by Orenef11** [comment]
@ShlomiBalalis 
Did you check it?
If so, can you attach the link JOB run?

File: sdcm/cluster.py (hunk 2 above)
Comment (ShlomiBalalis) [reply sdcm/cluster.py:1271]: I used the variable two lines below....

File: mgmt_upgrade_test.py (hunk 1 above)
Comment (ShlomiBalalis) [reply mgmt_upgrade_test.py:113]: I thought about that, but considering that message is not printed anywhere, it's a useless effort. The message is strictly for clarity when reading the code, not for the output.

**Comment by ShlomiBalalis** [comment]
> @ShlomiBalalis
> Did you check it?
> If so, can you attach the link JOB run?

https://jenkins.scylladb.com/job/manager-master/job/staging/job/centos-upgrade-test-staging/2

File: sdcm/cluster.py (hunk 2 above)
Comment (Orenef11) [reply sdcm/cluster.py:1271]: Sorry, I missed it...

**Review by Orenef11 — APPROVED** [review]
LGTM

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**